### PR TITLE
Users/profile

### PR DESCRIPTION
**Pull request by TruddyTheDuddi**
In this sprint, I've explored how the graphics side of Android Studio worked, created a concept design of what the **profile page** could look like, and built the base of that profile page for a user in xml.

I've created a **user class** that will serve as data holder for a user's information, and filled with a future database request. In this case the class contains the name, last name, username, date of join.

For now, only one user can be displayed on the profile, where the user's info is injected into the layout. For now it doesn't follow any View model, but in the next tasks this will be changed.

I've create a class for **pretty printing dates** to display user join date: instead of displaying a date it tells how much time elapsed since a specific date and today, such as "10 days ago".

**Issues during sprint**

- I had trouble choosing what Date class should be used. Kotlin's calendar would have been the best, but the minimum required API (23) doesn't support the feature (24). Can we change that? Java's newer Calendar and LocalTime class aren't supported either, so I stayed with the Java Date class, but it is deprecated and shouldn't be used.
- PrettyDate printing has a clumsy way of taking in a date for testing. Any improvement suggestions? @zaclegarssure @Gerussi

**TODO**

- [x] Add unit tests
- [x] Add android tests

### CODE REVIEW

**Comment by TruddyTheDuddi** [comment]
Also I've added the theme colors and fonts that should be used throughout the app if we keep this style

**Review by Gerussi — COMMENTED** [review]
in app/src/main/java/com/github/palFinderTeam/palfinder/profile/ProfileUser.kt wouldn't we need to store the picture reference into the class ProfileUser? For example, to display the picture of users that have joined a certain activity

**Comment by TruddyTheDuddi** [comment]
@Gerussi Yes, we will
The only reason I didn't code anything pictures related is because we need to see how we'll manage images in the app, do we need to store them in a folder of the app or are they directly accessible from the database, where do we store them, and how do we store them. It's definitely a bigger chunk of work.
I will have to do some research about that, it will be looked into in the next weeks

**Review by Gerussi — COMMENTED** [review]
[here](https://github.com/PalFinderTeam/palFinder/blob/f62b0a59e8661ff53dd18c4bd99d89fe69587882/app/src/main/java/com/github/palFinderTeam/palfinder/MainActivity.kt#L20) I think you commented a line that makes the test fail for the function sendMessage

File: app/src/main/java/com/github/palFinderTeam/palfinder/profile/ProfileUser.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+data class ProfileUser(
+    private val username: String,
+    private val name: String,
+    private val surname: String,
+    private val joinDate: Date
+) : Serializable {
+
+    fun getName(): String {
```
Comment (zaclegarssure) [inline app/src/main/java/com/github/palFinderTeam/palfinder/profile/ProfileUser.kt:19]: I think that kotlin has nice syntax shortcut for this kind of things. For instance if you specify the val without private, they are automatically gettable (not a real word sorry).

Comment (TruddyTheDuddi) [reply app/src/main/java/com/github/palFinderTeam/palfinder/profile/ProfileUser.kt]: Fixed, thanks

**Review by Gerussi — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -21,6 +21,9 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity
+            android:name=".ProfileActivity"
+            android:exported="false" />
```
Comment (zaclegarssure) [inline app/src/main/AndroidManifest.xml:26]: You may add a go back button (I think they call it upward navigation), in order to make the app demo easier.
Actually the go back button of the phone (emulator) works too so this is not much of an issue.

In any case navigation will pretty much change later on in the project so it is not a big deal.

**Review by zaclegarssure — APPROVED** [review]
Otherwise I think everything is fine. Good job !

**Pull request merged**